the_total_weekly_salary.rb

```diff
@@ -0,0 +1,67 @@
+=begin
+3. Write a function that takes an array of hours and returns the total weekly salary.
+The first element of the array is the number of hours spent at work on Monday.
+The second element of the array is the number of hours spent at work on Tuesday, etc.
+A worker earns $10 an hour for the first 8 hours.
+For every overtime hour, he's earning $15.
+On Saturday and Sunday, the employer pays twice for each hour and overtime.
+Examples
+weekly_salary([8, 8, 8, 8, 8, 0, 0]) ➞ 400
+weekly_salary([10, 10, 10, 0, 8, 0, 0]) ➞ 410
+weekly_salary([0, 0, 0, 0, 0, 12, 0]) ➞ 280
+Notes
+Every element in the array is greater than or equal to 0.
+=end
+
+def weekly_salary(array)
+  week_salary = 0
+  weekend_salary = 0
+  total_salary = 0
+  
+    #for week days 
+  
+  for i in 0..4

```

Comment:
why do this way? there is a lot of code that is duplicated between this and the next loop. Just extract the difference and put that in a condition inside the one loop? 

So you loop though the array you receive with index. The index is the day of the week. You use that to multiply in the end the total if that is a sat or the Sunday. 

then in each you just see what hours are standard rate, calculate extra rate and apply bonus for weekend. 

You can reduce this code by more than half.